Clockwise/ICircuitBreakerStorage.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System;
+
+namespace Clockwise
+{
+    public interface ICircuitBreakerStorage : IDisposable
+    {
+        event EventHandler<CircuitBreakerStateDescriptor> CircuitBreakerStateChanged;
```

Comment:
We shouldn't model this as an event. A subscribe/dispose pattern makes it a lot easier to intercept subscription and take action (e.g. connecting to remote services) based on that.